app/validators/open_before_closed_validator.rb

```diff
@@ -0,0 +1,15 @@
+class OpenBeforeClosedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, _value)
+    record.errors.add(attribute, error_message) unless before_closed?(record)
+  end
+
+private
+
+  def before_closed?(record)
+    record.opened_date <= record.closed_date || record.closed_date.blank?
+  end
```

Comment:
Hi,

If `opened_date` or `closed_date` are empty then it still works, I guess maybe it sends as an empty string in the form and then persists as nil? So it compares against `""` which works ok. I added some more feature tests... let me know if I am missing something!